homeassistant/components/ted/sensor.py

```diff
@@ -0,0 +1,218 @@
+            name="{name} Voltage",
+            native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+            device_class=SensorDeviceClass.VOLTAGE,
+        ),
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up TED sensor platform."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = data[COORDINATOR]
+    config_name = data[NAME]
+
+    entity_registry = hass.helpers.entity_registry.async_get(hass)
+    config_id = str(config_entry.unique_id)
+    entities = []
+    for desc in build_sensor_descs(config_name, "consumption", "Energy Usage", False):
+        entities.append(TedSensor(desc, config_id, coordinator))
+
+    if coordinator.data["type"] != SystemType.NET:
+        for desc in build_sensor_descs(config_name, "net", "Net Grid Energy", True):
+            entities.append(TedSensor(desc, config_id, coordinator))
+        for desc in build_sensor_descs(
+            config_name, "production", "Energy Production", False
+        ):
+            entities.append(TedSensor(desc, config_id, coordinator))
+
+    option_entities = []
+
+    for spyder_id, spyder in coordinator.data["spyders"].items():
+        for sensor_description in build_sensor_descs(
+            spyder["name"], "spyder_energy", "Energy Usage", False
+        ):
+            option_entities.append(
+                TedBreakdownSensor(
+                    "spyders",
+                    spyder_id,
+                    sensor_description,
+                    config_id,
+                    coordinator,
+                )
+            )
+
+    for mtu_id, mtu in coordinator.data["mtus"].items():
+        is_net = False
+        if mtu["type"] == MtuType.LOAD:
+            stype = "Energy Usage"
+        elif mtu["type"] == MtuType.GENERATION:
+            stype = "Energy Production"
+        else:
+            stype = "Net Grid Energy"
+            is_net = True
+        for sensor_description in build_mtu_sensor_descs(mtu["name"], stype, is_net):
+            option_entities.append(
+                TedBreakdownSensor(
+                    "mtus",
+                    mtu_id,
+                    sensor_description,
+                    config_id,
+                    coordinator,
+                )
+            )
+
+    for sensor in option_entities:
+        option = sensor.entity_description.key
+        if config_entry.options.get(option, OPTION_DEFAULTS[option]):
+            entities.append(sensor)
+        else:
+            entity_id = entity_registry.async_get_entity_id(
+                Platform.SENSOR, DOMAIN, sensor.unique_id
+            )
+            if entity_id:
+                _LOGGER.debug("Removing entity: %s", sensor.unique_id)
+                entity_registry.async_remove(entity_id)
+
+    async_add_entities(entities)
+
+
+class TedSensor(CoordinatorEntity, SensorEntity):
```

Comment:
For new integrations, we require the following:
- Support for [the new entity naming](https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations)
- Services must be [translatable](https://developers.home-assistant.io/blog/2023/07/11/translating-services)
- Entities and their attributes should be [translatable](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/)

Please adopt your code to add support for it.